pandas/_libs/internals.pyx

```diff
@@ -74,59 +75,65 @@ cdef class BlockPlacement:
         cdef:
             slice s = self._ensure_has_slice()
             Py_ssize_t start, stop, step, _
+
         if s is not None:
             start, stop, step, _ = slice_get_indices_ex(s)
             return iter(range(start, stop, step))
-        else:
-            return iter(self._as_array)
+
+        return iter(self._as_array)
 
     @property
     def as_slice(self) -> slice:
         cdef:
             slice s = self._ensure_has_slice()
-        if s is None:
-            raise TypeError('Not slice-like')
-        else:
+
+        if s is not None:
             return s
 
+        raise TypeError("Not slice-like")
+
     @property
     def indexer(self):
         cdef:
             slice s = self._ensure_has_slice()
+
         if s is not None:
             return s
-        else:
-            return self._as_array
```

Comment:
@jorisvandenbossche This is a very similar change to the change that was mentioned [here](https://github.com/pandas-dev/pandas/pull/32801#discussion_r394308161), would you like me to revert that as well?

Reply:
To be clear, my opinion is only a single one, but yes, I would personally revert this as well